File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/1036/BusinessApplication/Models/User.partial.vb
Imports System
Imports System.ComponentModel

Namespace Web

    Partial Public Class User
#Region "Rendre DisplayName Bindable"

        ''' <summary>
        ''' Substitution de la méthode <c>OnPropertyChanged</c> qui génère des notifications de modification de propriété lorsque <see cref="User.DisplayName"/> change.
        ''' </summary>
        ''' <param name="e">Arguments de l'événement de modification de la propriété.</param>
        Protected Overrides Sub OnPropertyChanged(ByVal e As System.ComponentModel.PropertyChangedEventArgs)
            MyBase.OnPropertyChanged(e)

            If e.PropertyName = "Name" Or e.PropertyName = "FriendlyName" Then
                Me.RaisePropertyChanged("DisplayName")
            End If
        End Sub
#End Region
    End Class
End Namespace

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/1036/BusinessApplication/Models/User.partial.cs
namespace $safeprojectname$.Web
{
    using System;
    using System.ComponentModel;

    /// <summary>
    /// Extensions de la classe <see cref="User"/>.
    /// </summary>
    public partial class User
    {
        #region Rendre DisplayName Bindable

        /// <summary>
        /// Substitution de la méthode <c>OnPropertyChanged</c> qui génère des notifications de modification de propriété lorsque <see cref="User.DisplayName"/> change.
        /// </summary>
        /// <param name="e">Arguments de l'événement de modification de la propriété.</param>
        protected override void OnPropertyChanged(PropertyChangedEventArgs e)
        {
            base.OnPropertyChanged(e);

            if (e.PropertyName == "Name" || e.PropertyName == "FriendlyName")
            {
                this.RaisePropertyChanged("DisplayName");
            }
        }
        #endregion
    }
}
